recipes/cpython/all/conanfile.py

```diff
@@ -512,6 +512,70 @@ def _msvc_package_copy(self):
         lib_dir_path = os.path.join(self.package_folder, self._msvc_install_subprefix, "Lib").replace("\\", "/")
         self.run(f"{interpreter_path} -c \"import compileall; compileall.compile_dir('{lib_dir_path}')\"")
 
+    def _exact_lib_name(self, folder):
+        possible_extensions = ("a", "so", "dylib", "lib")
+        for file in os.listdir(folder):
+            for extension in possible_extensions:
+                if re.match(f".*\\.{extension}", file):
+                    return file
+        raise ConanException(f"No library files found in {folder}")
+
+    @property
+    def _cmake_module_path(self):
+        if is_msvc(self):
+            # On Windows, `lib` is for Python modules, `libs` is for compiled objects.
+            # Usually CMake modules are packaged with the latter.
+            return os.path.join(self._msvc_install_subprefix, "libs", "cmake")
+        else:
+            return os.path.join("lib", "cmake")
+
+    def _write_cmake_findpython_wrapper_file(self):
+        template = textwrap.dedent("""
+        if (DEFINED Python3_VERSION_STRING)
+            set(_CONAN_PYTHON_SUFFIX "3")
+        else()
+            set(_CONAN_PYTHON_SUFFIX "")
+        endif()
+        set(Python${{_CONAN_PYTHON_SUFFIX}}_EXECUTABLE {})
+        set(Python${{_CONAN_PYTHON_SUFFIX}}_LIBRARY {})
```

Comment:
Using the standard Python formatter with this template is a bit of an overkill and bug-prone due to the numerous braces.
I would simplify the template and use
```py
content = template.replace("@PYTHON_EXE@", python_exe).replace("@PYTHON_LIBRARY@", python_library)
```

Reply:
Yea, I think I like that better. In my initial prototypes I didn't have that many braces, and then I realized I had to handle the difference between FindPython/FindPython3.

Reply:
New implementation pushed